features/graph/documentation/partner-onboarding.md

```diff
@@ -0,0 +1,312 @@
+
+The token endpoint should generally return a JSON-encoded result such as the snippet below:
+
+```json
+{
+   "access_token": "SlAV32hkKG",
+   "token_type": "Bearer",
+   "refresh_token": "8xLOxBtZp8",
+   "expires_in": 3600,
+   "id_token": "eyJhbGciOiJSUzI1NiIsImtpZCI6IjFlOWdkazcifQ.ewogImlzcyI6ICJodHRwOi8vc2VydmVyLmV4YW1wbGUuY29tIiwKICJzdWIiOiAiMjQ4Mjg5NzYxMDAxIiwKICJhdWQiOiAiczZCaGRSa3F0MyIsCiAibm9uY2UiOiAibi0wUzZfV3pBMk1qIiwKICJleHAiOiAxMzExMjgxOTcwLAogImlhdCI6IDEzMTEyODA5NzAKfQ.ggW8hZ1EuVLuxNuuIJKX_V8a_OMXzR0EHR9R6jgdqrOOF4daGU96Sr_P6qJp6IcmD3HP99Obi1PRs-cwh3LO-p146waJ8IhehcwL7F09JdijmBqkvPeB2T9CJNqeGpe-gccMg4vfKjkM8FcGvnzZUN4_KSP0aAp1tOJ1zZwgjxqGByKHiOtX7TpdQyHE5lcMiKPXfEIQILVq0pc_E2DzL7emopWoaoZTF_m0_N0YzFC6g6EJbOEoRoS     K5hoDalrcvRYLSrQAZZKflyuVCyixEoV9GfNQC3_osjzw2PAithfubEEBLuVVk4XUVrWOLrLl0nx7RkKU8NXNHq-rvKMzqg"
+  } 
+```
+
+#### UserInfo endpoint
+
+The `userInfo` response should, at a minimum, contain the `sub` value. It can also contain other relevant details for an application, such as name, email, avatar URL, or other data. An example of how the type would look like:
+
+```typescript
+type UserInfoResponse = {
+  /**
+  * The id of the authenticated user (the 'subject')
+  */
+  sub: string
+}
+```
+
+Example `userInfo` response that contains the required `sub` field as well as some additional (optional) user information:
+
+```json
+{
+   "sub": "248289761001",
+   "name": "Jane Doe",
+   "given_name": "Jane",
+   "family_name": "Doe",
+   "preferred_username": "j.doe",
+   "email": "janedoe@example.com",
+   "picture": "http://example.com/janedoe/me.jpg"
+ }
+ ```
+
+### Tokens (refresh, etc.)
+
+We expect **access tokens** to be relatively short-lived (on the order of hours to days, not months or infinite) and to be able to refresh them after they expire using a **one-time refresh token**. Upon refreshing, we expect a new access token and a new refresh token.
+
+### Scope description
+
+To show the user a friendly interface to understand what authentication is available in the API, we need a list of scopes with the following structure:
+
+```typescript
+type Scope = {
+  /**
+  * Actual programmatic scope specified during OAuth dance
+  */
+  scope: string;
+  /**
+  * Description as to the purpose of the scope / what it grants access to
+  */
+  description: string;
+  /**
+  * Human-friendly title of the scope
+  */
+  title: string;
+  /**
+  * A category to group multiple scopes under
+  */
+  category: string;
+  /**
+  * Whether the scope should be specified by default
+  */
+  isDefault: string;
+  /**
+  * Whether the scope is required to access the service
+  * If true, `isDefault` must be true.
+  *
+  * If this scope is required in order to retrieve the
+  * logged-in user information associated with this token
+  * (primarily their id in your service), then this field
+  * must be true.
+  */
+  isRequired: string;
+}
+```
+
+For example, here is how Spotify outlines the scopes in their own API:
```

Comment:
Do we need to list all of these? Are just a few enough to get the idea across?

Reply:
Good point - I shortened it.